backbone.js

```diff
@@ -159,11 +109,9 @@
     trigger: function(name) {
       if (!this._events) return this;
       var args = slice.call(arguments, 1);
-      if (!eventsApi(this, 'trigger', name, args)) return this;
-      var events = this._events[name];
-      var allEvents = this._events.all;
-      if (events) triggerEvents(events, args);
-      if (allEvents) triggerEvents(allEvents, arguments);
+
+      // Use `eventsApi` to normalize `name` into the proper event names.
```

Comment:
The bits with `triggerSentinel` and the numerous internal functions mostly. 

It almost seems like we should drop the `Events = {on: ..., off: ..., listenTo: ...}` object, and just colocate the different events methods (`Events.on = ...; onApi = ...; Events.off = ...;`.) To follow this logic, I'm having to jump back and forth through the source to even understand where the different pieces are going.

There's so much extra overhead and indirection going on here, it really detracts from the readability of the Events code.